chain/vm/runtime.go

```diff
@@ -440,8 +440,8 @@ func (ssh *shimStateHandle) Transaction(obj vmr.CBORer, f func()) {
 
 	c := ssh.rt.Put(obj)
 
-	// TODO: handle error below
-	ssh.rt.stateCommit(baseState, c)
+	err = ssh.rt.stateCommit(baseState, c)
+	panic(fmt.Errorf("failed to commit state after transaction: %w", err))
```

Comment:
I'm not sure if this is the correct thing to do, but panicing seems strictly better than dropping the error.